### PR TITLE
Added Command Buffer wrapper class

### PR DESCRIPTION
**Pull request by Raelr**
### Description
This PR adds a wrapper class for the Vulkan Command Buffer concept. This PR is being used to prepare the repo for the Renderer refactor needed for testures.

### The PR has been...
- [x] provided a reasonable name that is not just the branch name (e.g "Added Vulkan render delegate")
- [x] linked to its related issue
- [x] assigned a reviewer from the team
- [x] labelled appropriately

### The code has been...
- [x] made mergable and free of conflicts in relation to `master` *(according to GitHub)*
- [ ] pulled to the reviewer's machine and reasonably tested

### CODE REVIEW

File: engine/render/renderer/platform/vulkan/Swapchain.h [hunk 1]
```diff
@@ -96,7 +97,7 @@ class Swapchain
      * @param imageIndex - the index of the image being drawn
      * @return VkResult - the result of submitting the buffer
      */
-    Utils::Result SubmitCommandBuffers(const VkCommandBuffer* buffers, uint32_t* imageIndex);
+    Utils::Result SubmitCommandBuffers(CommandBuffer& commandBuffer, uint32_t imageIndex);
```
Comment (J-Mo63) [inline engine/render/renderer/platform/vulkan/Swapchain.h:101]: Why is this non-const?

File: engine/render/renderer/platform/vulkan/Swapchain.h [hunk 2]
```diff
@@ -167,11 +168,11 @@ class Swapchain
         return swapchainExtent;
     }
 
-    void BeginRenderPass(VkCommandBuffer commandBuffer,
+    void BeginRenderPass(Vulkan::CommandBuffer& commandBuffer,
                          uint32_t imageIndex,
                          std::initializer_list<VkClearValue> clearValues);
 
-    void EndRenderPass(VkCommandBuffer commandBuffer);
+    void EndRenderPass(Vulkan::CommandBuffer& commandBuffer);
```
Comment (J-Mo63) [inline engine/render/renderer/platform/vulkan/Swapchain.h:175]: Likewise for these

File: engine/render/renderer/platform/vulkan/Renderer.cpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
+//
+// Created by Aryeh Zinn on 13/10/2022.
+//
+
+#include "Renderer.h"
+
+namespace Siege
+{
+namespace Vulkan
+{} // namespace Vulkan
```
Comment (J-Mo63) [inline engine/render/renderer/platform/vulkan/Renderer.cpp:10]: ??

File: engine/render/renderer/platform/vulkan/Renderer.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+namespace Siege
+{
+namespace Vulkan
+{
+
+class Renderer
+{};
+
+} // namespace Vulkan
+} // namespace Siege
```
Comment (J-Mo63) [inline engine/render/renderer/platform/vulkan/Renderer.h:17]: ??

File: engine/render/renderer/platform/vulkan/CommandBuffer.h [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    inline VkCommandBuffer& operator[](const size_t idx)
+    {
+        return commandBuffers[idx];
+    }
+
+    /**
+     * Returns the raw Vulkan CommandBuffer pointer
+     * @param index the index of the CommandBuffer to be accessed
+     * @return
+     */
```
Comment (J-Mo63) [inline engine/render/renderer/platform/vulkan/CommandBuffer.h:77]: incomplete description, should also contain warning about out of bounds access

**Review by J-Mo63 — APPROVED** [review]
Runs fine, looks good. Few comments otherwise.

File: engine/render/renderer/platform/vulkan/Swapchain.h (hunk 2 above)
Comment (Raelr) [reply engine/render/renderer/platform/vulkan/Swapchain.h:175]: These change the CommandBuffer, so they can't be `const`.

File: engine/render/renderer/platform/vulkan/Swapchain.h (hunk 1 above)
Comment (Raelr) [reply engine/render/renderer/platform/vulkan/Swapchain.h:101]: We can safely make this `const`. Changed the function.

File: engine/render/renderer/platform/vulkan/Renderer.h (hunk 4 above)
Comment (Raelr) [reply engine/render/renderer/platform/vulkan/Renderer.h:17]: Removed.

File: engine/render/renderer/platform/vulkan/Renderer.cpp (hunk 3 above)
Comment (Raelr) [reply engine/render/renderer/platform/vulkan/Renderer.cpp:10]: Removed.

File: engine/render/renderer/platform/vulkan/CommandBuffer.h (hunk 5 above)
Comment (Raelr) [reply engine/render/renderer/platform/vulkan/CommandBuffer.h:77]: Adjusted.

**Pull request merged**